sdk/cosmos/azure-cosmos/test/test_crud_async.py

```diff
@@ -0,0 +1,2671 @@
+
+    async def initialize_client_with_connection_core_retry_config(self, retries):
+        start_time = time.time()
+        try:
+            cosmos_client.CosmosClient(
+                "https://localhost:9999",
+                CRUDTests.masterKey,
+                "Session",
+                retry_total=retries,
+                retry_read=retries,
+                retry_connect=retries,
+                retry_status=retries)
+            self.fail()
+        except AzureError as e:
+            end_time = time.time()
+            return end_time - start_time
+
+    async def test_absolute_client_timeout(self):
+        with self.assertRaises(exceptions.CosmosClientTimeoutError):
+            cosmos_client.CosmosClient(
+                "https://localhost:9999",
+                CRUDTests.masterKey,
+                "Session",
+                retry_total=3,
+                timeout=1)
+
+        error_response = ServiceResponseError("Read timeout")
+        timeout_transport = TimeoutTransport(error_response)
+        client = cosmos_client.CosmosClient(
+            self.host, self.masterKey, "Session", transport=timeout_transport, passthrough=True)
+
+        with self.assertRaises(exceptions.CosmosClientTimeoutError):
+            await client.create_database_if_not_exists("test", timeout=2)
+
+        status_response = 500  # Users connection level retry
+        timeout_transport = TimeoutTransport(status_response)
+        client = cosmos_client.CosmosClient(
+            self.host, self.masterKey, "Session", transport=timeout_transport, passthrough=True)
+        with self.assertRaises(exceptions.CosmosClientTimeoutError):
+            await client.create_database("test", timeout=2)
+
+        databases = client.list_databases(timeout=2)
+        with self.assertRaises(exceptions.CosmosClientTimeoutError):
+            databases = [database async for database in databases]
+
+        status_response = 429  # Uses Cosmos custom retry
+        timeout_transport = TimeoutTransport(status_response)
+        client = cosmos_client.CosmosClient(
+            self.host, self.masterKey, "Session", transport=timeout_transport, passthrough=True)
+        with self.assertRaises(exceptions.CosmosClientTimeoutError):
+            await client.create_database_if_not_exists("test", timeout=2)
+
+        databases = client.list_databases(timeout=2)
+        with self.assertRaises(exceptions.CosmosClientTimeoutError):
+            databases = [database async for database in databases]
+
+    async def test_query_iterable_functionality(self):
+        async def __create_resources(client):
+            """Creates resources for this test.
+
+            :Parameters:
+                - `client`: cosmos_client_connection.CosmosClientConnection
+
+            :Returns:
+                dict
+
+            """
+            collection = self.configs.create_multi_partition_collection_with_custom_pk_if_not_exist(self.client)
```

Comment:
this needs to be a new async method - any method in the self.configs is going to be for the sync client only
I would just do
```suggestion
            collection = await self.databaseForTest.create_container(<CONFIGS>)
```
where CONFIGS are the actual name and partition key being used in the original method in the configs file